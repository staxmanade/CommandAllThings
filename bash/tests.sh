
DIR="$( cd "$( dirname "$0" )" && pwd )"

# include the sample aliases
. $DIR/sampleProfile.sh

commands=( grunt gulp jake rake gradle make )

for i in "${commands[@]}"
do

	testDir=$(cd "$DIR/../samples/${i}Sample"; pwd)

	echo "*********************************"
	echo "*** $i"
	echo "*** Expected Testing Dir: $testDir"

	cd $testDir

	echo "*** Current Directory : $(pwd)"
	
    for j in "${commands[@]}"
    do
		if [ -e "package.json" ]; then
            npm install > /dev/null
		fi
		echo "*** Executing: $j"
	    eval $j
    done
done

