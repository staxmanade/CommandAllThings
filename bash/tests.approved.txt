*********************************
*** grunt
*** Expected Testing Dir: /Users/jasonjarrett/code/personal/CommandAllThings/samples/gruntSample
*** Current Directory : /Users/jasonjarrett/code/personal/CommandAllThings/samples/gruntSample
npm WARN package.json gruntSample@0.0.0 No description
npm WARN package.json gruntSample@0.0.0 No repository field.
npm WARN package.json gruntSample@0.0.0 No README data
*** Executing: grunt
[4mRunning "log:bar" (log) task[24m
Hello Grunt

[32mDone, without errors.[39m
npm WARN package.json gruntSample@0.0.0 No description
npm WARN package.json gruntSample@0.0.0 No repository field.
npm WARN package.json gruntSample@0.0.0 No README data
*** Executing: gulp
[4mRunning "log:bar" (log) task[24m
Hello Grunt

[32mDone, without errors.[39m
npm WARN package.json gruntSample@0.0.0 No description
npm WARN package.json gruntSample@0.0.0 No repository field.
npm WARN package.json gruntSample@0.0.0 No README data
*** Executing: jake
[4mRunning "log:bar" (log) task[24m
Hello Grunt

[32mDone, without errors.[39m
npm WARN package.json gruntSample@0.0.0 No description
npm WARN package.json gruntSample@0.0.0 No repository field.
npm WARN package.json gruntSample@0.0.0 No README data
*** Executing: rake
[4mRunning "log:bar" (log) task[24m
Hello Grunt

[32mDone, without errors.[39m
npm WARN package.json gruntSample@0.0.0 No description
npm WARN package.json gruntSample@0.0.0 No repository field.
npm WARN package.json gruntSample@0.0.0 No README data
*** Executing: gradle
[4mRunning "log:bar" (log) task[24m
Hello Grunt

[32mDone, without errors.[39m
npm WARN package.json gruntSample@0.0.0 No description
npm WARN package.json gruntSample@0.0.0 No repository field.
npm WARN package.json gruntSample@0.0.0 No README data
*** Executing: make
[4mRunning "log:bar" (log) task[24m
Hello Grunt

[32mDone, without errors.[39m
*********************************
*** gulp
*** Expected Testing Dir: /Users/jasonjarrett/code/personal/CommandAllThings/samples/gulpSample
*** Current Directory : /Users/jasonjarrett/code/personal/CommandAllThings/samples/gulpSample
npm WARN package.json gulpSample@0.0.0 No description
npm WARN package.json gulpSample@0.0.0 No repository field.
npm WARN package.json gulpSample@0.0.0 No README data
*** Executing: grunt
[gulp] Using gulpfile ~/code/personal/CommandAllThings/samples/gulpSample/gulpfile.js
[gulp] Starting 'test'...
Hello Gulp
[gulp] Finished 'test' after ##.# Î¼s
[gulp] Starting 'default'...
[gulp] Finished 'default' after ##.# Î¼s
npm WARN package.json gulpSample@0.0.0 No description
npm WARN package.json gulpSample@0.0.0 No repository field.
npm WARN package.json gulpSample@0.0.0 No README data
*** Executing: gulp
[gulp] Using gulpfile ~/code/personal/CommandAllThings/samples/gulpSample/gulpfile.js
[gulp] Starting 'test'...
Hello Gulp
[gulp] Finished 'test' after ##.# Î¼s
[gulp] Starting 'default'...
[gulp] Finished 'default' after ##.# Î¼s
npm WARN package.json gulpSample@0.0.0 No description
npm WARN package.json gulpSample@0.0.0 No repository field.
npm WARN package.json gulpSample@0.0.0 No README data
*** Executing: jake
[gulp] Using gulpfile ~/code/personal/CommandAllThings/samples/gulpSample/gulpfile.js
[gulp] Starting 'test'...
Hello Gulp
[gulp] Finished 'test' after ##.# Î¼s
[gulp] Starting 'default'...
[gulp] Finished 'default' after ##.# Î¼s
npm WARN package.json gulpSample@0.0.0 No description
npm WARN package.json gulpSample@0.0.0 No repository field.
npm WARN package.json gulpSample@0.0.0 No README data
*** Executing: rake
[gulp] Using gulpfile ~/code/personal/CommandAllThings/samples/gulpSample/gulpfile.js
[gulp] Starting 'test'...
Hello Gulp
[gulp] Finished 'test' after ##.# Î¼s
[gulp] Starting 'default'...
[gulp] Finished 'default' after ##.# Î¼s
npm WARN package.json gulpSample@0.0.0 No description
npm WARN package.json gulpSample@0.0.0 No repository field.
npm WARN package.json gulpSample@0.0.0 No README data
*** Executing: gradle
[gulp] Using gulpfile ~/code/personal/CommandAllThings/samples/gulpSample/gulpfile.js
[gulp] Starting 'test'...
Hello Gulp
[gulp] Finished 'test' after ##.# Î¼s
[gulp] Starting 'default'...
[gulp] Finished 'default' after ##.# Î¼s
npm WARN package.json gulpSample@0.0.0 No description
npm WARN package.json gulpSample@0.0.0 No repository field.
npm WARN package.json gulpSample@0.0.0 No README data
*** Executing: make
[gulp] Using gulpfile ~/code/personal/CommandAllThings/samples/gulpSample/gulpfile.js
[gulp] Starting 'test'...
Hello Gulp
[gulp] Finished 'test' after ##.# Î¼s
[gulp] Starting 'default'...
[gulp] Finished 'default' after ##.# Î¼s
*********************************
*** jake
*** Expected Testing Dir: /Users/jasonjarrett/code/personal/CommandAllThings/samples/jakeSample
*** Current Directory : /Users/jasonjarrett/code/personal/CommandAllThings/samples/jakeSample
*** Executing: grunt
*** Executing: gulp
*** Executing: jake
*** Executing: rake
*** Executing: gradle
*** Executing: make
*********************************
*** rake
*** Expected Testing Dir: /Users/jasonjarrett/code/personal/CommandAllThings/samples/rakeSample
*** Current Directory : /Users/jasonjarrett/code/personal/CommandAllThings/samples/rakeSample
*** Executing: grunt
Hello rake
*** Executing: gulp
Hello rake
*** Executing: jake
Hello rake
*** Executing: rake
Hello rake
*** Executing: gradle
Hello rake
*** Executing: make
Hello rake
*********************************
*** gradle
*** Expected Testing Dir: /Users/jasonjarrett/code/personal/CommandAllThings/samples/gradleSample
*** Current Directory : /Users/jasonjarrett/code/personal/CommandAllThings/samples/gradleSample
*** Executing: grunt
*** Executing: gulp
*** Executing: jake
*** Executing: rake
*** Executing: gradle
*** Executing: make
*********************************
*** make
*** Expected Testing Dir: /Users/jasonjarrett/code/personal/CommandAllThings/samples/makeSample
*** Current Directory : /Users/jasonjarrett/code/personal/CommandAllThings/samples/makeSample
*** Executing: grunt
hello make
*** Executing: gulp
hello make
*** Executing: jake
hello make
*** Executing: rake
hello make
*** Executing: gradle
hello make
*** Executing: make
hello make
